### PR TITLE
Set Up Sqlx and DB Schemas For Excretor

### LINKED ISSUE #9: Setup sqlx, schemas and db connection

### PR DESCRIPTION
**Pull request by harshkhandeparkar**
Closes #9

### CODE REVIEW

**Review by Devansh-bit — COMMENTED** [review]
the error handling can be done the rust way - [generic error handling in Axum](https://github.com/tokio-rs/axum/blob/main/examples/anyhow-error-response/src/main.rs)

Also the benefit of using `sqlx` over an ORM is the compile-time checked queries, using the `query!` macro. See [`sqlx::query!`](https://docs.rs/sqlx/latest/sqlx/macro.query.html) and [this](https://github.com/launchbadge/sqlx?tab=readme-ov-file#sqlx-is-not-an-orm)

File: .env.template [hunk 1]
```diff
@@ -2,3 +2,6 @@
 TUMMY_USERNAME=
 TUMMY_PASSWORD=
 TUMMY_PORT=5432
+
+# The port on which the excretor runs
+EXCRETOR_PORT=
```
Comment (chirag-ghosh) [inline .env.template:7]: Add a port by default

**Review by chirag-ghosh — CHANGES_REQUESTED** [review]

File: excretor/src/env.rs
Comment (YoganshSharma) [inline excretor/src/env.rs:1]: _Suggestion_: We can use [clap](https://docs.rs/clap/latest/clap/_derive/index.html) here. `cargo add clap --features=derive,env`
```suggestion
use clap::Parser;

#[derive(Parser)]
pub struct EnvVars {
    #[arg(env)]
    pub excretor_port: String,
    #[arg(env)]
    pub tummy_username: String,
    #[arg(env)]
    pub tummy_db: String,
    #[arg(env, default_value="3000")]
    pub tummy_port: String,
    #[arg(env)]
    pub tummy_host: String,
    #[arg(env)]
    pub tummy_password: String,
}
```
In the `main.rs` file use `EnvVars::parse()` to get EnvVars

**Review by YoganshSharma — CHANGES_REQUESTED** [review]

File: .env.template (hunk 1 above)
Comment (harshkhandeparkar) [reply .env.template:7]: Done.

File: excretor/src/env.rs
Comment (harshkhandeparkar) [reply excretor/src/env.rs:1]: Sure.

Comment (harshkhandeparkar) [reply excretor/src/env.rs:1]: Done.

**Comment by harshkhandeparkar** [comment]
> the error handling can be done the rust way - [generic error handling in Axum](https://github.com/tokio-rs/axum/blob/main/examples/anyhow-error-response/src/main.rs)

I think we should do that in a separate PR.

> Also the benefit of using `sqlx` over an ORM is the compile-time checked queries, using the `query!` macro. See [`sqlx::query!`](https://docs.rs/sqlx/latest/sqlx/macro.query.html) and [this](https://github.com/launchbadge/sqlx?tab=readme-ov-file#sqlx-is-not-an-orm)

As I mentioned on Slack as well, the `query!` macro requires a test database to check errors at compile time, and we'll need to set something up for that.

File: excretor/src/main.rs [hunk 2]
```diff
@@ -89,7 +134,7 @@ async fn get_messages(
             templates::ChannelPageTemplate {
                 messages,
                 page: pagination.page,
-                channel: Channel { name: channel },
+                channel: Channel { name: channel, topic: "Dear frontend developers. I am sorry you have to go through this but that's just how it is.".into(), purpose: "You are either a frontend developer cursing me right now or a backend developer. Neither case is good but either is necessary at the moment.".into() },
```
Comment (chirag-ghosh) [inline excretor/src/main.rs:137]: :clap:

File: excretor/src/main.rs [hunk 3]
```diff
@@ -75,10 +110,20 @@ async fn get_messages(
         messages.push(Message {
             id: i as i32,
             text: format!("Test message {}", i + 1),
+            channel_name: "idk yet".into(),
+            user_id: "A user.".into(),
+            ts: "0".into(),
+            thread_ts: "0".into(),
+            parent_user_id: "Another user".into(),
             user: User {
+                real_name: "Lmao the frontend devs are going to have a field day with this.".into(),
```
Comment (chirag-ghosh) [inline excretor/src/main.rs:119]: :clap:

File: excretor/src/main.rs [hunk 4]
```diff
@@ -75,10 +110,20 @@ async fn get_messages(
...
+                display_name: "Seriously though, the db has not yet been integrated here, so please bare with me.".into(),
+                email: "backend_developer@hell.com".into(),
```
Comment (chirag-ghosh) [inline excretor/src/main.rs:121]: :1st_place_medal:

File: excretor/src/main.rs [hunk 5]
```diff
@@ -36,27 +56,42 @@ struct Pagination {
...
+    }
+}
+
 async fn load_channel(Path(channel): Path<String>) -> (StatusCode, Response) {
     (
         StatusCode::OK,
         Html(
             templates::ChannelTemplate {
-                channel: Channel { name: channel },
+                channel: Channel { name: channel, purpose: "Nothing rn".into(), topic: "Hey if you are the frontend developer, make sure to ask someone to return the actual data.".into() },
```
Comment (chirag-ghosh) [inline excretor/src/main.rs:94]: :mechanical_arm:

**Review by chirag-ghosh — APPROVED** [review]
Docker part looks fine to me

File: excretor/src/models.rs [hunk 6]
```diff
@@ -3,18 +3,32 @@ use sqlx::prelude::FromRow;
...
+    pub channel_name: String,
+    pub user_id: String,
     pub text: String,
-    pub user: User,
+    // Timestamp
+    pub ts: String,
+    // Thread timestamp
+    pub thread_ts: String,
+    // If it is a thread, id of the user who sent the parent message
+    pub parent_user_id: String,
```
Comment (Devansh-bit) [inline excretor/src/models.rs:14]: Since this can be NULL for threads, it should be an `Option<String>` instead

**Review by Devansh-bit — CHANGES_REQUESTED** [review]

File: excretor/src/models.rs (hunk 6 above)
Comment (harshkhandeparkar) [reply excretor/src/models.rs:14]: I believe it will be empty, not null.

**Review by Devansh-bit — APPROVED** [review]

**Review by chirag-ghosh — APPROVED** [review]

**Pull request merged**